RecoEgamma/EgammaElectronProducers/plugins/TrackingRegionsFromSuperClustersProducer.h

```diff
@@ -247,15 +250,11 @@ getVtxPos(const edm::Event& iEvent,double& deltaZVertex)const
 {
   if(useZInVertex_){
     auto verticesHandle = getHandle(iEvent,verticesToken_);   
-    if(!verticesHandle->empty()){
-      deltaZVertex = originHalfLength_;
-      const auto& pv = verticesHandle->front();
-      return GlobalPoint(pv.x(),pv.y(),pv.z());
-    }
+    deltaZVertex = originHalfLength_;
+    const auto& pv = verticesHandle->front();
+    return GlobalPoint(pv.x(),pv.y(),pv.z());
```

Comment:
I'd still put back a protection for (pathological) events in which no vertex is reconstructed.
Using the beam spot in that case (as it was in the original code) was not a bad idea